controllers/ovndbcluster_controller.go

```diff
@@ -786,6 +806,69 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 	return ctrl.Result{}, nil
 }
 
+// generateServiceConfigMaps - create create configmaps which hold service configuration
+func (r *OVNDBClusterReconciler) generateExternalConfigMaps(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *ovnv1.OVNDBCluster,
+	serviceName string,
+	envVars *map[string]env.Setter,
+) error {
+	// Create/update configmaps from templates
+	cmLabels := labels.GetLabels(instance, labels.GetGroupLabel(serviceName), map[string]string{})
+	encapType := ""
+
+	externalEndpoint, err := instance.GetExternalEndpoint()
+	if err != nil {
+		return err
+	}
+
+	ovnController, err := ovnv1.GetOVNController(ctx, h, instance.Namespace)
+	if err != nil {
+		return err
+	} else if ovnController != nil {
+		encapType = ovnController.Spec.ExternalIDS.OvnEncapType
+	}
+
+	externalTemplateParameters := make(map[string]interface{})
+	externalTemplateParameters["OVNRemote"] = externalEndpoint
+	if encapType != "" {
+		externalTemplateParameters["OVNEncapType"] = encapType
+	}
+
+	cms := []util.Template{
+		// EDP ConfigMap
+		{
+			Name:          "ovncontroller-config",
+			Namespace:     instance.Namespace,
+			Type:          util.TemplateTypeConfig,
+			InstanceType:  instance.Kind,
+			Labels:        cmLabels,
+			ConfigOptions: externalTemplateParameters,
+		},
+	}
+	return configmap.EnsureConfigMaps(ctx, h, instance, cms, envVars)
+}
+
+func (r *OVNDBClusterReconciler) deleteExternalConfigMaps(
+	ctx context.Context,
+	h *helper.Helper,
+	namespace string,
+) error {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ovncontroller-config",
```

Comment:
Before this PR, the name of the CM was constructed from the OVNController instance name + `-config` prefix. Now it's hardcoded. This will break in case OVNController CR is named differently from `ovncontroller`. Is it safe? What if a user names their CR differently?

Reply:
I think you're correct here, I'm not familiar with most of the code of openstack-operator, hence I don't know who creates the [openstackdataplaneservices.dataplane.openstack.org/ovn](https://logserver.rdoproject.org/openstack-periodic-container-antelope-centos9/review.rdoproject.org/rdo-jobs/master/periodic-adoption-multinode-to-crc-ceph/97f28e3/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/crs/openstackdataplaneservices.dataplane.openstack.org/ovn.yaml) and where the `ovncontroller-config` comes from. But either we find another way to get the name correct or it's another reason to split the config maps used (or remove the ovnEncapType as mentioned earlier)

Reply:
So, I asked on slack and currently we're creating this config map the wrong way, as the name of the config map used by edpm deployment is also a magic string. Openstack-operator is using this [file to create the OpenStackDataPlaneService](https://github.com/openstack-k8s-operators/openstack-operator/blob/main/config/services/dataplane_v1beta1_openstackdataplaneservice_ovn.yaml).